packages/components/src/Chips/InternalChipDismissible/InternalChipDismissible.tsx

```diff
@@ -17,8 +17,33 @@ export function InternalChipDismissible(props: InternalChipDismissibleProps) {
     handleChipRemove,
     handleCustomAdd,
     handleWrapperKeyDown,
+    setChipRef,
   } = useInternalChipDismissible(props);
 
+  const chipElementsRef = useRef<Map<string, HTMLElement | null>>(new Map());
+
+  // Effect to update refs when chips are rendered
+  useEffect(() => {
+    // Wait for the DOM to be updated
+    setTimeout(() => {
+      if (wrapperRef.current) {
+        const chipElements =
+          wrapperRef.current.querySelectorAll('[role="option"]');
+
+        // Update ref map
+        chipElementsRef.current.clear();
+        sortedVisibleChipOptions.forEach((chip, index) => {
+          const element = chipElements[index] as HTMLElement | undefined;
```

Comment:
this casting feels weird. why would we want to cast it as `undefined`? how or why are we iterating over chips that aren't valid?